x-pack/legacy/plugins/apm/server/lib/metrics/by_agent/java/index.ts

```diff
@@ -10,9 +10,10 @@ import { getNonHeapMemoryChart } from './non_heap_memory';
 import { getThreadCountChart } from './thread_count';
 import { getCPUChartData } from '../shared/cpu';
 import { getMemoryChartData } from '../shared/memory';
+import { GenericMetricsChart } from '../../transform_metrics_chart';
 
 export async function getJavaMetricsCharts(setup: Setup, serviceName: string) {
-  const charts = await Promise.all([
+  const charts: GenericMetricsChart[] = await Promise.all([
```

Comment:
same here, curious why we need it

Reply:
It was previously used here: https://github.com/elastic/kibana/blob/33c237e187156777f1344b49c4b8445658ade224/x-pack/legacy/plugins/apm/server/lib/metrics/get_metrics_chart_data_by_agent.ts#L11-L13

Removing it caused typescript errors - often this happens when a const like `type: 'my-type'` is converted to `type: string`. Re-adding the type solved it.

Reply:
If you find a way to remove it and still keep typescript happy I'll gladly do that.

Reply:
Yeah, removing this causes this error:

```
legacy/plugins/apm/public/hooks/useServiceMetricCharts.ts:14:3 - error TS2322: Type '[]' is not assignable to type '[{ title: string; key: string; yUnit: YUnit; totalHits: number; series: { title: string; key: string; type: ChartType; color: string; overallValue: number | null; data: { x: number; y: number | null; }[]; }[]; }, { ...; }, { ...; }, { ...; }, { ...; }] | [...]'.
  Type '[]' is missing the following properties from type '[{ title: string; key: string; yUnit: YUnit; totalHits: number; series: { title: string; key: string; type: ChartType; color: string; overallValue: number | null; data: { x: number; y: number | null; }[]; }[]; }, { ...; }]': 0, 1

14   charts: []
     ~~~~~~

  legacy/plugins/apm/server/lib/metrics/by_agent/java/index.ts:24:12
    24   return { charts };
                  ~~~~~~
    The expected type comes from property 'charts' which is declared here on type '{ charts: [{ title: string; key: string; yUnit: YUnit; totalHits: number; series: { title: string; key: string; type: ChartType; color: string; overallValue: number | null; data: { x: number; y: number | null; }[]; }[]; }, { ...; }, { ...; }, { ...; }, { ...; }]; } | { ...; }'

legacy/plugins/apm/server/lib/metrics/by_agent/java/index.ts:13:1 - error TS6133: 'GenericMetricsChart' is declared but its value is never read.

13 import { GenericMetricsChart } from '../../transform_metrics_chart';
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
```patch
diff --git a/x-pack/legacy/plugins/apm/public/hooks/useServiceMetricCharts.ts b/x-pack/legacy/plugins/apm/public/hooks/useServiceMetricCharts.ts
index 3480ede7ac..b1b0315b00 100644
--- a/x-pack/legacy/plugins/apm/public/hooks/useServiceMetricCharts.ts
+++ b/x-pack/legacy/plugins/apm/public/hooks/useServiceMetricCharts.ts
@@ -10,8 +10,10 @@ import { useUiFilters } from '../context/UrlParamsContext';
 import { useFetcher } from './useFetcher';
 import { PromiseReturnType } from '../../typings/common';
 
-const INITIAL_DATA: PromiseReturnType<typeof loadMetricsChartData> = {
-  charts: []
+type MetricsChartDataResponse = PromiseReturnType<typeof loadMetricsChartData>;
+
+const INITIAL_DATA = {
+  charts: [] as Array<MetricsChartDataResponse['charts'][0]>
 };
 
 export function useServiceMetricCharts(
diff --git a/x-pack/legacy/plugins/apm/server/lib/metrics/by_agent/default.ts b/x-pack/legacy/plugins/apm/server/lib/metrics/by_agent/default.ts
index 453ba614f2..f55d3320c7 100644
--- a/x-pack/legacy/plugins/apm/server/lib/metrics/by_agent/default.ts
+++ b/x-pack/legacy/plugins/apm/server/lib/metrics/by_agent/default.ts
@@ -7,13 +7,12 @@
 import { Setup } from '../../helpers/setup_request';
 import { getCPUChartData } from './shared/cpu';
 import { getMemoryChartData } from './shared/memory';
-import { GenericMetricsChart } from '../transform_metrics_chart';
 
 export async function getDefaultMetricsCharts(
   setup: Setup,
   serviceName: string
 ) {
-  const charts: GenericMetricsChart[] = await Promise.all([
+  const charts = await Promise.all([
     getCPUChartData(setup, serviceName),
     getMemoryChartData(setup, serviceName)
   ]);
diff --git a/x-pack/legacy/plugins/apm/server/lib/metrics/by_agent/java/index.ts b/x-pack/legacy/plugins/apm/server/lib/metrics/by_agent/java/index.ts
index 7ea1d0bf69..9fc9f3029d 100644
--- a/x-pack/legacy/plugins/apm/server/lib/metrics/by_agent/java/index.ts
+++ b/x-pack/legacy/plugins/apm/server/lib/metrics/by_agent/java/index.ts
@@ -10,10 +10,9 @@ import { getNonHeapMemoryChart } from './non_heap_memory';
 import { getThreadCountChart } from './thread_count';
 import { getCPUChartData } from '../shared/cpu';
 import { getMemoryChartData } from '../shared/memory';
-import { GenericMetricsChart } from '../../transform_metrics_chart';
 
 export async function getJavaMetricsCharts(setup: Setup, serviceName: string) {
-  const charts: GenericMetricsChart[] = await Promise.all([
+  const charts = await Promise.all([
     getCPUChartData(setup, serviceName),
     getMemoryChartData(setup, serviceName),
     getHeapMemoryChart(setup, serviceName),

```